### ISSUE TITLE
Expose run information as environmental variables

### ISSUE THREAD

**Issue by oyvindeide** [equinor/ert#4904]
We have users who are parsing the runpath to get metadata about the ert run. This is data that we have available in ert, and we should provide it in a more robust fashion,

Environment variables on client:
1) Experiment ID
1) Realization ID
1) Simulation mode

On compute:
1) Iteration number
1) Realization number
1) Runpath format string

**Comment by eivindjahren** [comment]
For any given user job we can add the following to the job config to provide this information on compute:

```
EXEC_ENV ERT_REALIZATION <IENS>
EXEC_ENV ERT_ITERATION <ITER>
EXEC_ENV ERT_RUNPATH <RUNPATH>
```

**Comment by oysteoh** [comment]
We could also consider having one env-variable where we dump a json-string containing various data to be consumed by the users.

**Comment by mortalisk** [comment]
What is the intended goal here? Do we really want to help in parsing the runpath directly? I assume it is to get hold of certain data. Isn't it better to provide a way to ask for the data needed, in stead of encouraging monkeying about in the runpath?

**Comment by oysteoh** [comment]
> What is the intended goal here? Do we really want to help in parsing the runpath directly? I assume it is to get hold of certain data. Isn't it better to provide a way to ask for the data needed, in stead of encouraging monkeying about in the runpath?

If i understand  correctly - they _are_ currently using the runpath directly to deduce metadata. This issue is to expose the data in a more sane way, i.e through env-variable / one env-variable containing json.

**Comment by mortalisk** [comment]
Yes, but, that information will only help them in digging into the runpath, which ideally is an implementation detail anyways. They should be asking i.e. for a url to the file "example/file.DATA" that belongs to the realization 12, iteration 2, instead of the runpath.

**Comment by oysteoh** [comment]
> Yes, but, that information will only help them in digging into the runpath, which ideally is an implementation detail anyways. They should be asking i.e. for a url to the file "example/file.DATA" that belongs to the realization 12, iteration 2, instead of the runpath.

That is not what they are currently doing. They are extracting metadata from the runpath to upload with other data ( sumo are currently doing this ). Retrieving metadata / context from the runpath is not ideal - thus they want us to provide them with the possibility of getting these values from an _api_, i.e environment variables.

**Comment by eivindjahren** [comment]
We discussed this briefly in person and came to the conclusion that changing the default `ext_job.exec_env` to `{"_ERT_ITERATION_NUMBER": "<ITER>","_ERT_REALIZATION_NUMBER": "<IENS>", "_ERT_RUNPATH": "<RUNPATH>"}` is probably safe as it just modifies exiting functionality slightly.

**Comment by oysteoh** [comment]
This needs also to be implemented for workflows, i.e as the issue describes we have implemented for forward models ( compute ) but not workflows ( client )

- [ ] Environment variables on client:

    Experiment ID
    Realization ID
    Simulation mode

- [x] On compute:

    Iteration number
    Realization number
    Runpath format string
